hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -361,6 +366,19 @@ protected synchronized void scanInternal(Option<KeySpec> keySpecOpt) {
     }
   }
 
+  /**
+   * Check whether the current instants are valid
+   *
+   * @param instantTime
+   * @return
+   */
+  private boolean checkIfValidCommit(String instantTime) {
+    HoodieTimeline deltaCommitTimeline = this.hoodieTableMetaClient.getActiveTimeline().getDeltaCommitTimeline().filterCompletedInstants();
+    return deltaCommitTimeline.containsInstant(instantTime)
+        || (deltaCommitTimeline.isBeforeTimelineStarts(instantTime)
+        && this.hoodieTableMetaClient.getArchivedTimeline().getDeltaCommitTimeline().filterCompletedInstants().containsOrBeforeTimelineStarts(instantTime));
```

Comment:
`getArchivedTimeline()` loads all instants in the archived timeline by deserializing the commit metadata stored in Avro, which is time-consuming.  Can you avoid calling that if possible?

Reply:
For the case wrt rollback described in HUDI-3644, there should be a ROLLBACK_PREVIOUS_BLOCK that invalidates the previous delta commit log block.  Could you verify if the fix is actually needed?  It would be good to write a reproducible test to demonstrate the issue.